core/src/main/scala/org/http4s/MediaType.scala

```diff
@@ -176,320 +187,176 @@ sealed class MediaType(
+    new MediaType(
+      "multipart",
+      subType,
+      MimeDB.Compressible,
+      MimeDB.NotBinary,
+      Nil,
+      extensions = ext)
   }
 
-  private[this] def app(
-      subType: String,
-      compressible: Boolean,
-      binary: Boolean,
-      fileExtensions: String*) =
-    registerValue(new MediaType("application", subType, compressible, binary, fileExtensions))
-
-  private[this] def aud(subType: String, compressible: Boolean, fileExtensions: String*) =
-    registerValue(new MediaType("audio", subType, compressible, binary, fileExtensions))
-
-  private[this] def img(
-      subType: String,
-      compressible: Boolean,
-      binary: Boolean,
-      fileExtensions: String*) =
-    registerValue(new MediaType("image", subType, compressible, binary, fileExtensions))
-
-  private[this] def msg(subType: String, fileExtensions: String*) =
-    registerValue(new MediaType("message", subType, compressible, notBinary, fileExtensions))
-
-  private[this] def txt(subType: String, fileExtensions: String*) =
-    registerValue(new MediaType("text", subType, compressible, notBinary, fileExtensions))
-
-  private[this] def vid(subType: String, fileExtensions: String*) =
-    registerValue(new MediaType("video", subType, uncompressible, binary, fileExtensions))
-
-  // scalastyle:off line.size.limit
-  val `application/atom+xml` = app("atom+xml", compressible, notBinary, "atom")
-  val `application/base64` = app("base64", compressible, binary, "mm", "mme")
-  val `application/excel` = app(
-    "excel",
-    uncompressible,
-    binary,
-    "xl",
-    "xla",
-    "xlb",
-    "xlc",
-    "xld",
-    "xlk",
-    "xll",
-    "xlm",
-    "xls",
-    "xlt",
-    "xlv",
-    "xlw")
-  val `application/font-woff` = app("font-woff", uncompressible, binary, "woff")
-  val `application/gnutar` = app("gnutar", uncompressible, binary, "tgz")
-  val `application/gzip` = app("gzip", uncompressible, binary, "gz")
-  val `application/hal+json` = app("hal+json", compressible, binary) // we treat JSON as binary, since its encoding is not variable but defined by RFC4627
-  val `application/java-archive` = app("java-archive", uncompressible, binary, "jar", "war", "ear")
-  val `application/javascript` = app("javascript", compressible, notBinary, "js")
-  val `application/json` = app("json", compressible, binary, "json") // we treat JSON as binary, since its encoding is not variable but defined by RFC4627
-  val `application/lha` = app("lha", uncompressible, binary, "lha")
-  val `application/lzx` = app("lzx", uncompressible, binary, "lzx")
-  val `application/mspowerpoint` =
-    app("mspowerpoint", uncompressible, binary, "pot", "pps", "ppt", "ppz")
-  val `application/msword` =
-    app("msword", uncompressible, binary, "doc", "dot", "w6w", "wiz", "word", "wri")
-  val `application/octet-stream` = app(
+  /////////////////////////// PREDEFINED MEDIA-TYPE DEFINITION ////////////////////////////
+  // Copied from the definitions on MimeDB
```

Comment:
Predefined `MediaTypes` are copied verbatim from `MimeDB`

Reply:
I don't understand what's here vs. in MimeDB.

Reply:
I struggled with this a bit. We had several `MediaTypes` pinned here and now those are in `MimeDB`. There are a few alternatives:

1. We remove these altogether and make callers do e.g. `MimeDB.text.text/plain`
2. In `MediaType` we create types like `text/plain` and link them to `MimeDB.text.text/plain`
3. Or we copy the definitions from `MimeDB` to here (The option used)

Ideally, we'd use one of the first options but it gets a bit complex with `application`. There are a few thousand `application` mime types, and given the JVM code size limits it needs to be split on the code generation part. e.g. `application/xml` is `MimeDB.application.application_2.application/xml`. Even worse the numbering is not fixed as the mimedb can change so an update of `MimeDB` could break code. In option 1 that's not acceptable as we'd require users to change theyr code. option 2 maybe better as we'd need to update the links when `MimeDB` changes anyway.

I'm not heavily invested in option 3. we could switch to option 2 if you think is better